armi/reactor/converters/uniformMesh.py

```diff
@@ -269,32 +286,254 @@ def makeAssemWithUniformMesh(sourceAssem, newMesh):
+                    if paramName not in cachedParams[destBlock]:
+                        continue
+
+                    if isinstance(destBlock.p[paramName], numpy.ndarray):
+                        # Using just all() on the list/array is not sufficient because if a zero value exists
+                        # in the data then this would then lead to overwritting the data. This is an edge case see
+                        # in the testing, so this excludes zero values on the check.
+                        if (
+                            not all(
+                                [val for val in destBlock.p[paramName] if val != 0.0]
+                            )
+                            or not destBlock.p[paramName].size > 0
+                        ):
+                            destBlock.p[paramName] = cachedParams[destBlock][paramName]
+                    elif isinstance(destBlock.p[paramName], list):
+                        if (
+                            not all(
+                                [val for val in destBlock.p[paramName] if val != 0.0]
+                            )
+                            or not destBlock.p[paramName]
+                        ):
+                            destBlock.p[paramName] = cachedParams[destBlock][paramName]
+                    else:
+                        if not destBlock.p[paramName]:
+                            destBlock.p[paramName] = cachedParams[destBlock][paramName]
+
+    @staticmethod
+    def clearStateOnAssemblies(
+        assems, blockScalarParamNames=None, blockArrayParamNames=None, cache=True
+    ):
+        """
+        Clears the parameter state of blocks for a list of assemblies.
+
+        Parameters
+        ----------
+        assems : List[`Assembly <armi.reactor.assemblies.Assembly>`]
+            List of assembly objects.
+        blockScalarParamNames : List[str], optional
+            A list of block parameter names that can be assigned scalar values (i.e., int/float/string).
+            Examples: `flux`, `power`. If this is not provided then no scalar parameters will be mapped.
+        blockArrayParamNames : List[str], optional
+            A list of block parameter names that can be assigned array values (i.e., numpy.array/list).
+            Example: `mgFlux`. If this is not provided then no array parameters will be mapped.
+        cache : bool
+            If True, the block parameters that were cleared are stored
+            and returned as a dictionary of ``{b: {param1: val1, param2: val2}, b2: {...}, ...}``
+        """
+        if blockScalarParamNames is None:
+            blockScalarParamNames = []
+        if blockArrayParamNames is None:
+            blockArrayParamNames = []
+
+        cachedBlockParamData = collections.defaultdict(dict)
+
+        blocks = []
+        for a in assems:
+            blocks.extend(a.getBlocks())
+        for b in blocks:
+            for scalarParam in blockScalarParamNames:
+                if cache:
```

Comment:
i think this code is same for scalor and array params so you could iterate the 2 lists together if you want to save lines of code.